relay-dynamic-config/src/metrics.rs

```diff
@@ -638,6 +643,39 @@ pub fn convert_conditional_tagging(project_config: &mut ProjectConfig) {
     }
 }
 
+/// Configuration for metric extrapolation from sampled data.
+#[derive(Clone, Debug, Default, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct ExtrapolationConfig {
+    /// A list of MRI glob patterns to include in extrapolation.
+    #[serde(default)]
+    pub include: Vec<LazyGlob>,
+
+    /// A list of MRI glob patterns to exclude from extrapolation, overriding inclusion.
+    #[serde(default)]
+    pub exclude: Vec<LazyGlob>,
+}
+
+impl ExtrapolationConfig {
+    /// Returns `true` if this config is empty.
+    pub fn is_empty(&self) -> bool {
+        self.include.is_empty()
+    }
+
+    /// Returns `true` if the given metric resource identifier matches the include and exclude
+    /// patterns.
+    pub fn matches(&self, mri: &str) -> bool {
+        !self
+            .exclude
+            .iter()
+            .any(|glob| glob.compiled().is_match(mri))
+            && self
+                .include
+                .iter()
+                .any(|glob| glob.compiled().is_match(mri))
```

Comment:
nit: Could change the order here, if it's not included in the include we don't have to check excludes.

```suggestion
        self
            .include
            .iter()
            .any(|glob| glob.compiled().is_match(mri))
            && !self
                .exclude
                .iter()
                .any(|glob| glob.compiled().is_match(mri))
```

Reply:
@jjbayer had suggested the opposite during implementation, hence the order. `exclude` is likely to have fewer and the cheaper globs, so we can execute those first.